src/ai/backend/manager/idle.py

```diff
@@ -307,24 +305,133 @@ class UtilizationIdleChecker(BaseIdleChecker):
     """
 
     name: ClassVar[str] = "utilization"
-    default_cpu_util_threshold: ClassVar[float] = 30.0
-    default_accelerator_util_threshold: ClassVar[float] = 10.0
-    default_initial_grace_period: ClassVar[float] = 300.0  # allow first 5 minutes to be idle
+    cpu_util_series: List[float] = []
+    mem_util_series: List[float] = []
 
-    async def populate_config(self, config: Mapping[str, Any]) -> None:
-        pass
+    idle_timeout: timedelta
 
-    async def check_session(self, session: Row, dbconn: SAConnection) -> bool:
-        # last_stat = session['last_stat']
-        # TODO: implement
-        return True
+    _config_iv = t.Dict(
+        {
+            t.Key("threshold", default="10m"): tx.TimeDuration(),
+        }
+    ).allow_extra("*")
+
+    _evhandlers: List[EventHandler[None, AbstractEvent]]
+
+    async def populate_config(self, raw_config: Mapping[str, Any]) -> None:
+        config = self._config_iv.check(raw_config)
+        self.idle_timeout = config["threshold"]
+        self.cpu_thresh = int(raw_config["resource-thresholds"]["cpu"]["average"])
+        self.mem_thresh = int(raw_config["resource-thresholds"]["mem"]["average"])
+        self.wind = raw_config["resource-thresholds"]["cpu"]["window"]
+
+        log.info(
+            f"ExecutionIdleChecker: default cpu idle_timeout = {self.cpu_thresh} pct; \
+              Mem idle execution = {self.mem_thresh} %, \
+              and observed moving window = {self.wind}"
+        )
 
     async def update_app_streaming_status(
         self,
         session_id: SessionId,
         status: AppStreamingStatus,
     ) -> None:
-        pass
+        if status == AppStreamingStatus.HAS_ACTIVE_CONNECTIONS:
+            await self._disable_timeout(session_id)
+        elif status == AppStreamingStatus.NO_ACTIVE_CONNECTIONS:
+            await self._update_timeout(session_id)
+
+    async def _disable_timeout(self, session_id: SessionId) -> None:
+        log.debug(f"ExecutionIdleChecker._disable_timeout({session_id})")
+        await self._redis.set(
+            f"session.{session_id}.last_access", "0", exist=self._redis.SET_IF_EXIST
+        )
+
+    async def _update_timeout(self, session_id: SessionId) -> None:
+        log.debug(f"ExecutionIdleChecker._update_timeout({session_id})")
+        t = await self._redis.time()
+        await self._redis.set(
+            f"session.{session_id}.last_access",
+            f"{t:.06f}",
+            expire=max(86400, int(self.idle_timeout.total_seconds() * 2)),
```

Comment:
`idle_timeout` does not seem to be a valid property name since the utilization checker does not garbage collect compute session by the `idle_timeout`. Here, it is just a check period for the utilization.